### PR TITLE
Add an indexing property

### PR DESCRIPTION
**Pull request by ghost**
For Hyrax, it would be super useful to have indexing specified in the model. This PR adds an optional property ('indexing'), with a controlled list of values.

### CODE REVIEW

File: m3.yml [hunk 1]
```diff
@@ -127,6 +128,10 @@ properties:
       minimum: 0
       maximum: 100
     index_documentation: "searchable, displayable, creator facet"
+    indexing:
+      - "_tesim" # stored searchable
```
Comment (no-reply) [inline m3.yml]: tying these to blacklight style suffixes seems fragile, to me.

could we define a small indexing vocabulary (even just the text values you have in comments seem workable), and let applications map them as appropriate?

Comment (geekscruff) [reply m3.yml]: Yes, that makes sense. Will work on it soon!

**Comment by geekscruff** [comment]
@no-reply I've added a simpler human-friendly indexing list

**Review by arwenhutt — COMMENTED** [review]
@geekscruff This looks like it lines up with the groups discussions about documenting indexing and seems to be a useful addition!

Though I'm a little confused about the values to use -- possibly just my limited knowledge of solr -- but there seems to be a mix of the application neutral terms (e.g. displayable, facetable) and the solr abbreviations ("_tesim", "_sim".) in the updates to the json schema and the m3.yaml file.

**Comment by geekscruff** [comment]
@arwenhutt sorry, that was a mistake, I've updated all the indexing fields to match the vocab now (in m3.yml, others will need updating)

**Comment by arwenhutt** [comment]
@geekscruff great, thanks!  I think this looks good so am going to go ahead and merge.

**Pull request merged**